prow/spyglass/spyglass.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package spyglass creates views for Prow job artifacts
+package spyglass
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/deck/jobs"
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+// Spyglass records which sets of artifacts need views for a prow job
+type Spyglass struct {
+
+	// map of names of views to their corresponding lenses
+	Lenses map[string]Lens
+
+	// Job Agent for spyglass
+	Ja *jobs.JobAgent
+}
+
+// Lens is a single view of a set of artifacts
+type Lens struct {
+	// Name of the view, unique within a job
+	Name string
+	// Title of view
+	Title string
+	// Html rendering of the view
+	HTMLView string
+	// Priority is the relative position of the view on the page
+	Priority int
+}
+
+// ViewRequest holds data sent by a view
+type ViewRequest struct {
+	ViewName    string              `json:"name"`
+	ViewerCache map[string][]string `json:"viewerCache"`
+	ViewData    string              `json:"viewData"`
+}
+
+// NewSpyglass constructs a spyglass object from a JobAgent
+func NewSpyglass(ja *jobs.JobAgent) *Spyglass {
+	return &Spyglass{
+		Lenses: make(map[string]Lens),
+		Ja:     ja,
+	}
+}
+
+// Views gets all views of all artifact files matching each regexp with a registered viewer
+func (s *Spyglass) Views(eyepiece map[string][]string) []Lens {
+	lenses := []Lens{}
+	for viewer, matches := range eyepiece {
+		if len(matches) != 0 {
+			title, err := viewers.Title(viewer)
+			if err != nil {
+				logrus.Error("Could not find artifact viewer with name ", viewer)
+			}
+			priority, err := viewers.Priority(viewer)
+			if err != nil {
+				logrus.Error("Could not find artifact viewer with name ", viewer)
```

Comment:
Coule not find priority for viewer:

Reply:
`Could not find priority for viewer: `* :-)